pkg/fswatcher/fswatcher.go

```diff
@@ -104,245 +156,117 @@ func (w *Watcher) Close() {
 	close(w.stop)
```

Comment:
Would it be possible to make this wait for the `loop` to exit? Perhaps also worth closing the channels. This way code using this that also uses `goleak` in tests wouldn't be flaky.

Reply:
Done, thanks for the suggestion.